src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Controllers/DataSetsController.cs

```diff
@@ -250,4 +252,37 @@ public async Task<ActionResult<DataSetQueryPaginatedResultsViewModel>> QueryData
             .Query(dataSetId, request, dataSetVersion, cancellationToken)
             .HandleFailuresOrOk();
     }
+
+    /// <summary>
+    /// Download a data set as CSV
+    /// </summary>
+    /// <remarks>
+    /// The CSV response will render its metadata in a human-readable format (instead of
+    /// machine-readable IDs). The CSV is not subject to the same backward compatibility
+    /// guarantees as the data set's JSON representation in other endpoints.
+    /// </remarks>
+    [HttpGet("{dataSetId:guid}/csv")]
+    [Produces(MediaTypeNames.Text.Csv, MediaTypeNames.Application.Json)]
+    [SwaggerResponse(200, "The data set CSV file.")]
+    [SwaggerResponse(403, type: typeof(ProblemDetailsViewModel))]
+    [SwaggerResponse(404, type: typeof(ProblemDetailsViewModel))]
+    public async Task<ActionResult> DownloadDataSet(
```

Comment:
One last thing - can we change this to `DownloadDataSetCsv`?

Reply:
done :)